sources/packages/web/src/components/common/notes/CreateNoteModal.vue

```diff
@@ -49,51 +73,52 @@ export default {
       required: true,
     },
   },
-  emits: ["submitData"],
-  setup(props: any, context: any) {
-    const { showDialog, showModal } = useModalDialog<void>();
-    const formData = ref();
-    const formioUtils = useFormioUtils();
-    const dialogClosed = () => {
-      showDialog.value = false;
-    };
-    const formLoaded = async (form: any) => {
-      formData.value = form;
-      const dropdown = formioUtils.getComponent(form, "noteType");
-      const options = [{}];
+
+  setup(props: any) {
+    const { showDialog, showModal, resolvePromise } = useModalDialog<
+      NoteAPIInDTO | boolean
+    >();
+    const addNewNoteForm = ref({} as VForm);
+    const formModel = reactive({} as NoteAPIInDTO);
+    const noteTypeItems = ref([] as NoteTypeItemsDTO[]);
+
+    onMounted(async () => {
       if (props.entityType === NoteEntityType.Institution) {
         for (const noteType in InstitutionNoteType) {
-          options.push({ label: noteType, value: noteType });
+          noteTypeItems.value.push({ title: noteType, value: noteType });
         }
       }
       if (props.entityType === NoteEntityType.Student) {
         for (const noteType in StudentNoteType) {
-          options.push({ label: noteType, value: noteType });
+          noteTypeItems.value.push({ title: noteType, value: noteType });
         }
       }
-      dropdown.component.data.values = options;
-      dropdown.redraw();
-    };
-    const submitForm = async () => {
-      return formData.value.submit();
-    };
-    const submitNote = async (data: NoteAPIInDTO) => {
-      context.emit("submitData", data);
-    };
-    const addNewNote = async () => {
-      const formSubmitted = await submitForm();
-      if (formSubmitted) {
-        showDialog.value = false;
+    });
+
+    const submit = async () => {
+      const validationResult = await addNewNoteForm.value.validate();
+      if (!validationResult.valid) {
+        return;
       }
+      const payload = { ...formModel };
```

Comment:
We can add a comment here to describe why are we using a copy of payload instead of payload itself.

Reply:
I would like to know why are we using it actually @guru-aot?